sherpa/utils/tests/test_random.py

```diff
@@ -0,0 +1,165 @@
+#
+
+import numpy as np
+
+import pytest
+
+from sherpa.utils.random import poisson_noise, \
+    standard_normal, chisquare, choice
+from sherpa.utils.numeric_types import SherpaFloat
+
+
+def test_poisson_noise_checks_dtype():
+
+    with pytest.raises(ValueError,
+                       match="could not convert string to float: 'ham'"):
+        poisson_noise('ham')
+
+
+def test_poisson_noise_checks_number_args():
+
+    with pytest.raises(TypeError,
+                       match=r"^poisson_noise\(\) takes from 1 to 2 positional arguments but 3 were given$"):
+        poisson_noise(1, 2, 'ham')
+
+
+def test_poisson_noise_checks_rng():
+
+    with pytest.raises(AttributeError,
+                       match="'int' object has no attribute 'poisson'"):
+        poisson_noise(1, rng=123)
+
+
+def test_poisson_noise_scalar_none():
+    """Send in a scalar, get back a scalar"""
+
+    out = poisson_noise(1000)
+    assert type(out) == SherpaFloat
+    assert out > 0.0
+
+
+@pytest.mark.parametrize("x,expected", [(1000, 956), (4.3, 2)])
+def test_poisson_noise_scalar_rng(x, expected):
+    """Send in a scalar, get back a scalar"""
+
+    rng = np.random.RandomState(43)
+    out = poisson_noise(x, rng=rng)
+    assert type(out) == SherpaFloat
+    assert out == pytest.approx(expected)
+
+
+@pytest.mark.parametrize("x", [-1000, 0])
+@pytest.mark.parametrize("rng", [None, np.random.RandomState(23)])
+def test_poisson_noise_scalar_not_positive(x, rng):
+
+    out = poisson_noise(x, rng=rng)
+    assert type(out) == SherpaFloat
+    assert out == 0.0
+
+
+def test_poisson_noise_array_none():
+    """Send in an array, get back an array"""
+
+    out = poisson_noise([1001, 1002, 0.0, 1003, -1004])
+    assert type(out) == np.ndarray
+    assert out.dtype.type == SherpaFloat
+
+    # As the RNG is not set here, technically we could get zeros for
+    # the positive expected values, but this is very unlikely.
+    #
+    ans = np.flatnonzero(out > 0.0)
+    assert (ans == np.array([0, 1, 3])).all()
+
+
+def test_poisson_noise_array_rng():
+    """Send in an array, get back an array"""
+
+    rng = np.random.RandomState(43)
+    out = poisson_noise([12, 4.5, 0.0, 8.2, -1], rng=rng)
+    assert type(out) == np.ndarray
+    assert out.dtype.type == SherpaFloat
+
+    assert out == pytest.approx([7, 2, 0, 8, 0])
+
+
+def test_standard_normal_rng_is_none():
+    """Coverage shows this was not tested, so check it
+
+    It is not obvious if this is a real coverage failure,
+    or just an issue with how the tests are run, but may
+    as well add the check.
+    """
+
+    SEED = 1234
+    SIZE = 5
+    try:
+        np.random.seed(SEED)
+        rng = np.random.RandomState(SEED)
+        a = standard_normal(rng=None, size=SIZE)
+        b = standard_normal(rng=rng, size=SIZE)
+    finally:
+        np.random.seed()
+
+    assert a == pytest.approx(b)
+
+
+def test_chisquare_rng_is_none():
+    """Coverage shows this was not tested, so check it
```

Comment:
Similarly here. I suggest to decribe wha'ts tested, not why you added it and how ceverage works or doesn't work.

Reply:
ditto.